lavalink/client.py

```diff
@@ -203,11 +229,18 @@ def add_node(self, host: str, port: int, password: str, region: str,
         :param event:
             The event to dispatch to the hooks.
         """
-        for hook in self._event_hooks:
-            try:
+        unknown_events = self._event_hooks.get(None) or []
+        registered_events = self._event_hooks.get(event) or []
+
+        event_hooks = unknown_events + registered_events
+
+        try:
+            for hook in event_hooks:
                 if inspect.iscoroutinefunction(hook):
                     await hook(event)
                 else:
                     hook(event)
```

Comment:
@Devoxin @Zenrac should we really allow non-async funcs in here? And potentially holdup the whole thread and WS conn?

Reply:
It was non-async before so I kept it like that

Reply:
~~ya know that's why i mention those two~~

Reply:
Good point. Maybe remove support for non-async funcs. Can mark it as a breaking change

Reply:
Then `itertools.chain()` will not work.

Reply:
Can we not patch in our own version and put it in utils or something?
```py
def chain(*arrays):
  for a in arrays:
    for e in a:
      yield e
```

used like so:
```py
for hook in utils.chain(generic_hooks, targeted_hooks):
  await hook(event)
```